src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -197,6 +198,11 @@ internal DecisionTree ComputeDecisionTree()
                         else
                         {
                             AddToDecisionTree(result, label);
+                            if (label.Guard == null || label.Guard.ConstantValue == ConstantValue.True)
+                            {
+                                // Only unconditional switch labels contribute to subsumption
+                                AddToDecisionTree(subsumptionTree, label);
```

Comment:
No, we are maintaining separate decision trees for sybsumption checking (`subsumption`) and for code generation (`result`). Like it says on the tin.

---
In reply to: [79474816](https://github.com/dotnet/roslyn/pull/13770#discussion_r79474816) [](ancestors = 79474816)